SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialWorkerFlags.h

```diff
@@ -2,40 +2,38 @@
 
 #pragma once
 
-#include "Kismet/BlueprintFunctionLibrary.h"
 #include <WorkerSDK/improbable/c_worker.h>
 #include "SpatialWorkerFlags.generated.h"
 
-DECLARE_DYNAMIC_DELEGATE_TwoParams(FOnWorkerFlagsUpdatedBP, FString, FlagName, FString, FlagValue);
-DECLARE_DYNAMIC_MULTICAST_DELEGATE_TwoParams(FOnWorkerFlagsUpdated, FString, FlagName, FString, FlagValue);
+DECLARE_DYNAMIC_DELEGATE_TwoParams(FOnWorkerFlagsUpdatedBP, const FString&, FlagName, const FString&, FlagValue);
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_TwoParams(FOnWorkerFlagsUpdated, const FString&, FlagName, const FString&, FlagValue);
 
 UCLASS()
-class SPATIALGDK_API USpatialWorkerFlags : public UBlueprintFunctionLibrary
+class SPATIALGDK_API USpatialWorkerFlags : public UObject
 {
 	GENERATED_BODY()
 
 public:
 	/** Gets value of a worker flag. Must be connected to SpatialOS to properly work.
-	 * @param Name - Name of worker flag
-	 * @param OutValue - Value of worker flag
+	 * @param InFlagName - Name of worker flag
+	 * @param OutFlagValue - Value of worker flag
 	 * @return - If worker flag was found.
 	 */
-	UFUNCTION(BlueprintCallable, Category="SpatialOS")
-	static bool GetWorkerFlag(const FString& Name, FString& OutValue);
-	
-	static FOnWorkerFlagsUpdated& GetOnWorkerFlagsUpdated();
+	bool GetWorkerFlag(const FString& InFlagName, FString& OutFlagValue) const;
```

Comment:
Possible alternative as you basically only want flags as a read only thing and out params are the devil:
Could return a pair of bool, string and name it try get.

Reply:
As I just refactored the class,  that looks like a different task we need to add?